vizalerts.py

```diff
@@ -615,18 +616,38 @@ def process_csv(csvpath, view, sitename, viewname, subscriberemail, subscribersy
             data = get_unique_vizdata(data, has_consolidate_email, has_email_from, has_email_cc, has_email_bcc, has_email_header, has_email_footer)
             rowcount_unique = len(data)
 
-            # determine whether we need to render the viz image--if so, render it to file
-            vizimagerefs = find_vizimages_in_body(data, has_email_header, has_email_footer)
-            if vizimagerefs:
+            
+            # could be multiple viz images for a single row in the CSV
+            # return a list of all found VIZ_IMAGE() strings including any custom views w/ or w/out URL parameters
+            # leaving them as VIZ_IMAGE() strings because it makes the search & replace easier for the body & attachments
+            vizimagerefs = []
+            vizimagerefs = find_vizimages_in_body(data, viewurlsuffix, has_email_header, has_email_footer)
+
+            # storing view in original value because we'll be rewriting it in order to call
+            # tabhttp.export_view
+            origviewsuffix = view['view_url_suffix']
+
+            
+            # creating a dictionary of imagepaths with vizimageref as the key
+            imagepaths = dict()
+            for vizimageref in vizimagerefs:
+                if vizimageref == IMAGE_PLACEHOLDER:
+                    view['view_url_suffix'] = origviewsuffix
+                else:
+                    view['view_url_suffix'] = re.match(u'.*?\(\'(.*?)\'\)', vizimageref).group(1)
+                    logger.debug('~tmp ' + view['view_url_suffix'])
```

Comment:
Yes, anything with logger.debug('~[...] is one for my debugging, sorry for
leaving those in there!

On Wed, Mar 2, 2016 at 6:36 PM, Matt <notifications@github.com> wrote:

> In vizalerts.py
> <https://github.com/tableau/VizAlerts/pull/14#discussion_r54812408>:
>
> > +            vizimagerefs = []
> > +            vizimagerefs = find_vizimages_in_body(data, viewurlsuffix, has_email_header, has_email_footer)
> > +
> > +            # storing view in original value because we'll be rewriting it in order to call
> > +            # tabhttp.export_view
> > +            origviewsuffix = view['view_url_suffix']
> > +
> > +
> > +            # creating a dictionary of imagepaths with vizimageref as the key
> > +            imagepaths = dict()
> > +            for vizimageref in vizimagerefs:
> > +                if vizimageref == IMAGE_PLACEHOLDER:
> > +                    view['view_url_suffix'] = origviewsuffix
> > +                else:
> > +                    view['view_url_suffix'] = re.match(u'.*?\(\'(.*?)\'\)', vizimageref).group(1)
> > +                    logger.debug('~tmp ' + view['view_url_suffix'])
>
> tabhttp.export_view will log this...can we remove this logger line too?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tableau/VizAlerts/pull/14/files#r54812408>.
>